site/content/docs/contributing/project-structure.md

```diff
@@ -9,42 +9,139 @@ menu:
+
+
+[update-generated.sh](/kind/hack/update-generated.sh) in short, generates Go source code that is necessary to use a Kubernetes-style resource definition to define a schema that can be use to configure Kind.
+
+Going a bit more in depth, update-generated.sh Installs
+[go-bindata](https://github.com/jteeuwen/go-bindata)
+which is used to embed static assets in Go. It also installs the deepcopy-gen, defaulter-gen, and conversion-gen tools from
+[kubernetes/code-generator](https://github.com/kubernetes/code-generator).
+These programs  are used to generate Kubernetes-like APIs. These programs are run in the following sequence: deepcopy -> defaulter -> conversion.
+
+To understand this process better, we need to keep in mind that kind’s configuration schema dictates how to bootstrap a Kubernetes cluster. The schema is defined in
+[kind/pkg/cluster/config](/pkg/cluster/config).
+In this directory, currently, you will see two subdirectories:
+[v1beta1](/pkg/cluster/config/v1beta1) and
+[v1beta2](/pkg/cluster/config/v1beta2).
+Each of these subdirectories correspond to a version of kind’s cluster configuration.
+
+One of the concerns with versioned configurations is enabling the project to be compatible with all. With this in mind, [kind/pkg/cluster/config](/pkg/cluster/config) contains the main configuration fields while [v1beta1](/pkg/cluster/config/v1beta1) and [v1beta2](/pkg/cluster/config/v1beta2) build on top of it.
+
+The way this is implemented is by running deepcopy-gen and defaulter-gen in
+[kind/pkg/cluster/config](/pkg/cluster/config),
+followed by running deepcopy-gen, defaulter-gen, and conversion-gen on all version subdirectories (i.e., [v1beta1](/pkg/cluster/config/v1beta1) and [v1beta1](/pkg/cluster/config/v1beta1)).
+
+The [kubernetes/code-generator](https://github.com/kubernetes/code-generator) tools work by comment tags which are specified in the `doc.go` file for each directory. For example, all `doc.go` files within [kind/pkg/cluster/config](/pkg/cluster/config) have the following tags:
+```
+// +k8s:deepcopy-gen=package
+// +k8s:conversion-gen=sigs.k8s.io/kind/pkg/cluster/config
+// +k8s:defaulter-gen=TypeMeta
+```
+
+Additionally, [pkg/cluster/config/types.go](pkg/cluster/config/types.go) has an additional tag:
+```
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+```
+
+Let’s see how this tags work with the
+[kubernetes/code-generator](https://github.com/kubernetes/code-generator)
+binaries deepcopy-gen, defaulter-gen, and conversion-gen.
+
+For each of the directories related to defining a configuration for kind, we start by running [deepcopy-gen](https://godoc.org/k8s.io/code-generator/cmd/deepcopy-gen). deepcopy-gen generates functions that efficiently perform a full deep-copy of each type that is part of the configuration. These functions will be used to pass values to each type, go from a versioned configuration to the top level one.
```

Comment:
`These functions will be used to pass values to each type, go from a versioned configuration to the top level one.`

The deepcopy functions are more just used "throughout the codebase" imo - saying they are used to go from versions configuration to the top level one, whilst *technically* is probably the case, is a bit misleading as they aren't specifically for that (whereas conversion-gen is)

Reply:
You're right. Removed that sentence